src/test/java/shop/CartTest.java

```diff
@@ -0,0 +1,55 @@
+package shop;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CartTest {
+
+    private RealItem realItem;
+
+    private VirtualItem virtualItem;
+
+    @BeforeEach
+    @Tag("CartTests")
+    void initItems() {
+        VirtualItem virtualItem = new VirtualItem();
```

Comment:
your tests may be shorter. The general logic of testing the cart may be as follows:

```
class CartTest {

    private RealItem realItem;
    private Cart cart;

    @BeforeEach
    void initTestData() {
        cart = new Cart("testCart");
        realItem = new RealItem();
        realItem.setPrice(100);
        cart.addRealItem(realItem);
    }

    @Test
    void testCalculationAfterAddingRealItemToCart() {
        assertEquals(120.0, cart.getTotalPrice());
    }

    @Test
    void testCalculationAfterRemovingRealItemFromCart() {
        cart.deleteRealItem(realItem);
        assertEquals(0, cart.getTotalPrice());
    }
}
```